storybuilder-org/StoryCAD
# Fix how StoryCAD deals with onedrive files when they are unavailable.

This PR updates how storycad handles Cloud Storage Providers such as google drive/onedrive/dropbox etc.
StoryCAD had some partial code relating to OneDrive it's evident it doesn't function as intended.

Here's a indepth explanation of the new code:
StorageFiles have a property, StorageFile.IsAvailable which is great as it returns true if the user is online and the file is cached/downloaded or able to be downloaded. This does not take into account if the user has paused file syncing with the provider  meaning that pausing syncs within onedrive/dropbox would cause an issue if the file wasn't cached as it would return true even though it's not available as onedrive/dropbox/etc will not sync the file as it's been told not to.

So first we check the above property as if it does return false, theres no point continuing to load the file as it wont sync. So a flag is raised that the load failed due to a cloud error.

If this does return true, we call ReadAllText on the file, this will read the file. At this point one of two possibilitys can occur:
- Windows will realise the file needs to be synced and will pause StoryCAD to sync the file, if this happens the load will occur normally.
- An IO exception will occur because the user has paused syncing or some other cloud io related error has occured, at this point we verify the file has the offline attribute, which denotes it's cloud based file; a flag is raised that the load failed due to a cloud error.

Once the above tests are complete, StoryCAD will check if the load failed, if it has then a content dialog will open informing the user the load failed because the file is on a cloud storage providers server. The message is designed to be a clear as possible so that users are not at any stage worried about data loss and contains a link to a manual page explaining how to troubleshoot cloud storage providers.


Steps to test/reproduce:
1) Put outline file in OneDrive/Dropbox/etc.
2) Set file to not be saved locally.
3) disconnect from wifi/disable syncing
4) open file in storycad.


This PR removes the old onedrive check as its defective.
This PR will be a draft until manual changes are approved.
